amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/polling/PrefetchRecordsPublisher.java

```diff
@@ -511,6 +511,9 @@ private void makeRetrievalAttempt() {
                 } catch (InterruptedException e) {
                     Thread.currentThread().interrupt();
                     log.info("{} :  Thread was interrupted, indicating shutdown was called on the cache.", streamAndShardId);
+                } catch (InvalidArgumentException e) {
```

Comment:
This needs to be before RetryableRetrievalException or within that check for IAE, here it wont be executed. You should add a test to make sure this works.

Reply:
Im surprised the test passed I think there are 2 different InvalidArgumentException, can we make sure what is thrown by SDK and use that, maybe your code is correct, but lets make sure.